### PR TITLE
search: fix segfault

### PR DESCRIPTION
**Pull request by camdencheek**
The error streamed back was pointing to a nil error rather than the
error encountered, so we'd get an event with both a nil error and a nil
result.

For context, this was causing a segmentation fault further up the chain when we were trying to access the fields of `StreamSearchEvent.SearchResult`: https://sourcegraph.slack.com/archives/CMBA8F926/p1611088281013800

### CODE REVIEW

**Comment by uwedeportivo** [comment]
makes total sense. classic copy/paste :-)

**Review by uwedeportivo — APPROVED** [review]

**Review by rvantonder — APPROVED** [review]
thanks for writing a test!

sounds-reasonable

File: internal/search/backend/horizontal_test.go [hunk 1]
```diff
@@ -114,6 +114,31 @@ func TestHorizontalSearcher(t *testing.T) {
 	searcher.Close()
 }
 
+func TestHorizontalSearcherError(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (camdencheek) [reply internal/search/backend/horizontal_test.go:117]: Cool, sounds good to me. If you find that page, could you forward it on to me? I've always been a little hazy on good practices for test structure/naming.

[the remark this answers is not in the document]
Comment (rvantonder) [reply internal/search/backend/horizontal_test.go]: I found it: https://about.sourcegraph.com/handbook/engineering/testing#conventions

It's a bit misplaced, I will move it under https://about.sourcegraph.com/handbook/engineering/languages/testing_go_code and link to the Go page at the `engineering/testing` toplevel.

**Pull request merged**